### ISSUE TITLE
8.5.1 php 8.1.24 Massive Log Spam as below

### ISSUE THREAD

**Issue by antonical** [salesagility/SuiteCRM-Core#436]
Before you open an issue, please check if a similar issue already exists or has been closed before.

**I cannot see this issue.**

Provide a more detailed introduction to the issue itself, and why you consider it to be a bug

**The lines of the log provided below should be evidence of this.**

Expected Behavior
**The product should work without error for basic functions. I have no idea what the implications of this error are on the integrity of the overall installation. We are unable to upload images to products or change the site logo for instance.**

Actual Behavior
**The php Error log is spammed with these 4 repeating log lines on using the product. We expect to be able to upload images to products or change the site logo etc. We are still testing but this looks like a PHP incompatibility issue.**

Also please check relevant logs (suitecrm.log, php error.log etc.)
**suitecrm.log shows nothing while these issues are happening**

Possible Fix
**I am not a php developer so would have no idea.**

Steps to Reproduce
**Clean install on a Ubuntu 22.04 Virtualmin Server. php 8.1.27 enabled for the virtual server. All compatibility matrix conditions met. Normal operations of the platform results in a massive amount of php error log spam as below. The 4 log lines are repeated 000's of times and continually.**

```
php error log excerpt

[15-Feb-2024 09:26:03 UTC] PHP Deprecated:  Return type of Symfony\Component\HttpFoundation\Session\Attribute\AttributeBag::getIterator() should either be compatible with IteratorAggregate::getIterator(): Traversable, or the #[\ReturnTypeWillChange] attribute should be used to temporarily suppress the notice in /mydomain.com/public_html/crm/vendor/symfony/http-foundation/Session/Attribute/AttributeBag.php on line 134
[15-Feb-2024 09:26:03 UTC] PHP Deprecated:  Return type of Symfony\Component\HttpFoundation\Session\Attribute\AttributeBag::count() should either be compatible with Countable::count(): int, or the #[\ReturnTypeWillChange] attribute should be used to temporarily suppress the notice in /mydomain.com/public_html/crm/vendor/symfony/http-foundation/Session/Attribute/AttributeBag.php on line 144
[15-Feb-2024 09:26:03 UTC] PHP Deprecated:  Return type of Symfony\Component\HttpFoundation\Session\Session::getIterator() should either be compatible with IteratorAggregate::getIterator(): Traversable, or the #[\ReturnTypeWillChange] attribute should be used to temporarily suppress the notice in /mydomain.com/public_html/crm/vendor/symfony/http-foundation/Session/Session.php on line 131
[15-Feb-2024 09:26:03 UTC] PHP Deprecated:  Return type of Symfony\Component\HttpFoundation\Session\Session::count() should either be compatible with Countable::count(): int, or the #[\ReturnTypeWillChange] attribute should be used to temporarily suppress the notice in /mydomain.com/public_html/crm/vendor/symfony/http-foundation/Session/Session.php on line 141
Same 4 lines are repeated 1000’s of times.

Seems to be clear this is a php bug in the various .php files in the log.
```

Context
**We cannot do basic things like add an image to a product or change the site logo. We are still testing but I suspect this bug(s) are impacting the use of the platform in unexpected ways.
We would consider this high priority as the product does not function on php 8.1 or above.**

Your Environment
**SuiteCRM Version used: 8.5.1
Browser name and version tested various browsers this is a backend issue as we can see in the logs above
Environment name and version (e.g. MySQL, PHP 7): Apache 2.4.52php 8.1.27 MariaDB 10.6.16
Operating System and version (e.g Ubuntu 16.04): Ubuntu 22.04
See all software versions:
Apache version 2.4.52
PHP versions 7.4.33, 8.1.27, 8.3.2
MariaDB version 10.6.16**

**Comment by antonical** [comment]
@mattlorimer So you don't find this high priority that my logs in testing with a single user are filling up fast! I shudder to think how much log spam we would accumulate if we released it to users! 

Articles like this https://stackoverflow.com/questions/71133749/reference-return-type-of-should-either-be-compatible-with-or-the-re 

And a hundred more going back a number of years point to the sames issues. If there is a hot fix just publish it. It all seems to be related to symfony!

Cheers
Tony

**Comment by pgorod** [comment]
@antonical you are mixing a few things here. Deprecated messages are not errors. Not even a bit. They have zero meaning of error. They are essentially messages for devs to worry about in the near future. When they do worry about it, they do it "in bulk", sometimes there are hundreds of places to update the code when upgrading to a newer PHP version, but usually they're also easy to find and they don't require individual bug reports.

The issue with the images does appear to be a bug. But it's a different issue that would need a separate Github Issue with a different title and other kinds of information. Now that you have solved it (saw that in the forums) there are good clues that you can provide to make sure this gets looked at.

About the log spam - it's not happening on other people's servers. As far as we know, it could be something peculiar to your installation. That's why this is a case for the forums, not for Github. At least not yet.

**Comment by antonical** [comment]
@pgorod Yes I think I understand your point. Deprecated messages are going to turn into errors if they are not looked at and PHP versions continue to push forward. 

On the other issues around uploading of images I have opened another ticket and provided as much info as I can. I am not a php developer so who knows. Now we can upload images to products we are now testing other things. Email seems very flaky and complex in configuration to get anything to show up in the inbound email page. Despite inbound and outbounds configurations that all seemingly pass the connection tests. When it does by luck or magic import of emails os a problem. On some user accounts I can get a few emails imported on most despite selecting them and importing nothing shows up.

So onwards.......
